ci/scripts/build-with-meson.sh

```diff
@@ -65,16 +65,52 @@ function main() {
 
     pushd "${SANDBOX_DIR}"
 
-    show_header "Run test suite"
-    meson configure -Dtests=true -Db_coverage=true -Dipc=true
+    show_header "Run ASAN/UBSAN test suite"
```

Comment:
We might want to split these into separate jobs but punting that for now. I do think it would be nice to make the Meson job run as part of regular CI in a follow up if that's something you are open to.

Figure it doesn't hurt to have another tool to analyze the codebase, and I think Meson is a good deal easier to mess with than CMake. The downside of course is CMake isn't going anywhere anytime soon, so we do have active support for both